pkg/appfile/parser.go

```diff
@@ -406,3 +435,93 @@ func PrepareProcessContext(k8sClient client.Client, wl *Workload, applicationNam
 	}
 	return pCtx, nil
 }
+
+// GetOutputSecretNames set all secret names, which are generated by cloud resource, to context
+func GetOutputSecretNames(workloads *Workload) (string, error) {
+	secretName, err := getComponentSetting(process.OutputSecretName, workloads.Params)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprint(secretName), nil
+}
+
+func parseWorkloadK8sSecretSchema(ctx context.Context, c client.Client, namespace string, wl *Workload) ([]process.RequiredSecrets, error) {
+	var requiredSecret []process.RequiredSecrets
+	api, err := utils.GenerateOpenAPISchemaFromDefinition(wl.Name, wl.Template)
+	if err != nil {
+		if !errors.Is(err, errors.Errorf(utils.ErrNoSectionParameterInCue, wl.Name)) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	schema, err := utils.ConvertOpenAPISchema2SwaggerObject(api)
+	if err != nil {
+		return nil, err
+	}
+	tag := utils.K8sSecretSchemaTag
+	for k, v := range schema.Properties {
+		description := v.Value.Description
+		if strings.Contains(description, tag) {
+			contextName := strings.Split(description, tag)[1]
+			contextName = strings.TrimSpace(contextName)
+			secretNameInterface, err := getComponentSetting(k, wl.Params)
+			if err != nil {
+				return nil, err
+			}
+			secretName := fmt.Sprint(secretNameInterface)
```

Comment:
a formal way is assert it to string. And check if it's ok

```suggestion
			secretName,ok := secretNameInterface.(string)
```

Reply:
Great. But when should I use `func Sprint(a ...interface{}) string`?

Reply:
when you don't know the type